imgur.sh

```diff
@@ -45,15 +45,22 @@ while [ $# -gt 0 ]; do
 	file="$1"
 	shift
 
-	# Check file exists
-	if [ "$file" != "-" -a ! -f "$file" ]; then
-		echo "File '$file' doesn't exist; skipping" >&2
-		errors=true
-		continue
-	fi
-
 	# Upload the image
-	response=$(curl -s -H "Authorization: Client-ID $client_id" -H "Expect: " -F "image=@$file" https://api.imgur.com/3/image.xml) 2>/dev/null
+	# URL -> imgur
+	if [[ "$file" =~ ^https?://* ]]; then
+		response=$(curl -s -H "Authorization: Client-ID $client_id" -H "Expect: " -F "image=$file" https://api.imgur.com/3/image.xml) 2>/dev/null
+
+	# file -> imgur
+	else
+		# Check file exists
+		if [ "$file" != "-" -a ! -f "$file" ]; then
+			echo "File '$file' doesn't exist; skipping" >&2
+			errors=true
+			continue
+		fi
+
+		response=$(curl -s -H "Authorization: Client-ID $client_id" -H "Expect: " -F "image=@$file" https://api.imgur.com/3/image.xml) 2>/dev/null
+	fi
 	# The "Expect: " header is to get around a problem when using this through
 	# the Squid proxy. Not sure if it's a Squid bug or what.
```

Comment:
This comment is now awkwardly placed. Can you think of a way to refactor such that the `curl` command template is only set once, and then used in these two conditional branches? That way this comment can live with the template and we repeat less.